src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -660,159 +660,154 @@ private unsafe void DisposeInternal()
 
             var exceptionAggregator = new ExceptionAggregator(_logger, $"Could not dispose {nameof(DocumentDatabase)} {Name}");
 
-            var lockTaken = false;
-            Monitor.TryEnter(_clusterLocker, TimeSpan.FromSeconds(5), ref lockTaken);
+            // must acquire the lock in order to prevent concurrent access to index files
+            Monitor.Enter(_clusterLocker);
```

Comment:
Do we need to block it here to prevent the concurrent access?
Maybe if lock not taking try to take it again down the line?

Reply:
We must wait since this can cause concurrent access to the indexes.